File: python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self._minute = minute % 60
        self._hour = ((hour % 24) + (minute // 60 % 24)) % 24

    def __repr__(self):
        return f"{self._hour:02}:{self._minute:02}"

    def __eq__(self, other):
        return self.__repr__() == str(other)

    def __add__(self, minutes):
        self._clock_calc(minutes)
        return self

    def __sub__(self, minutes):
        self._clock_calc(minutes * -1)
        return self

    def _clock_calc(self, length):
        hour = self._hour + (length // 60)
        minute = self._minute + (length % 60)
        hour += minute // 60

        self._minute = minute % 60
        self._hour = hour % 24

File: go/clock/clock.go
package clock

import (
	"fmt"
)

type Clock struct {
	hour   int
	minute int
}

func New(hour, minute int) Clock {
	delta_h := minute / 60
	delta_m := minute % 60

	if delta_m < 0 {
		delta_h--
		minute = delta_m + 60
	}

	hour = (hour + delta_h) % 24

	if hour < 0 {
		hour += 24
	}

	minute %= 60

	return Clock{
		hour:   hour,
		minute: minute,
	}

}

func (c Clock) String() string {
	return fmt.Sprintf("%.2d:%.2d", c.hour, c.minute)
}

func (c Clock) Add(minutes int) Clock {
	return New(c.hour, c.minute+minutes)
}

func (c Clock) Subtract(minutes int) Clock {
	return New(c.hour, c.minute-minutes)
}
